spendsterIOS/Presentation/Expenses/CategorySelect/CategoryModel.swift

```diff
@@ -0,0 +1,73 @@
+//
+//  CategoryModel.swift
+//  spendsterIOS
+//
+//  Created by Dmytro Holovko on 5/10/19.
+//  Copyright © 2019 KeyToTech. All rights reserved.
+//
+
+import Foundation
+import RxSwift
+import Alamofire
+import SwiftyJSON
+
+class CategoryModel: CategoryModelProtocol {
+    private let url = "https://spendsterapp.herokuapp.com/categories"
+    func fetchCategories() -> Single<[CategoryResponse]> {
+        return Single<[CategoryResponse]>.create { single in
+            let request = Alamofire.request(self.url,
+                                            method: .get,
+                                            encoding: JSONEncoding.default)
+                .responseJSON { response in
+                    print(response)
+                    
+                    if let data = response.data,
+                        let json = try? JSON(data: data) {
+                        let categories = self.deserialize(json: json)
+                        single(.success(categories))
+                    } else if let error = response.error {
+                        single(.error(error))
+                    } else {
+                        single(.error(CategoryError.init(message: "")))
+                    }
+            }
+            return Disposables.create {
+                request.cancel()
+            }
+        }
+    }
+    
+    func deserialize(json: JSON) -> [CategoryResponse] {
```

Comment:
@Thongvor could be extracted from here into a separate class

Reply:
done

Reply:
removed